src/protocol/tx-validity.md

```diff
@@ -205,13 +293,42 @@ Once the free balances are computed, the [script is executed](../fuel-vm/index.m
 1. The unspent free balance `unspentBalance` for each asset ID.
 1. The unspent gas `unspentGas` from the `$ggas` register.
 
-The fees incurred for a transaction are `ceiling(((size(tx) * GAS_PER_BYTE) + (tx.gasLimit - unspentGas) + sum(tx.inputs[i].predicateGasUsed)) * tx.gasPrice / GAS_PRICE_FACTOR)`.
-
-`size(tx)` includes the entire transaction serialized according to the transaction format, including witness data.
+`size(tx)` encompasses the entire transaction serialized according to the transaction format, including witness data.
 This ensures every byte of block space either on Fuel or corresponding DA layer can be accounted for.
 
 If the transaction as included in a block does not match this final transaction, the block is invalid.
 
+### Fees
+
+The cost incurred by a transaction can be calculated by:
+
+```py
+cost(tx) = gas_to_fee(gas_cost(tx) - unspentGas, tx.gasPrice)
+```
+
+where:
+
+- `gas_cost(tx)` is the final cost of the transaction in gas, including gas fees incurred from:
+  - The number of bytes comprising the transaction
+  - Processing inputs and outputs
+  - VM initialization
+  - Predicate and script execution
+- `unspentGas` is the amount gas left over after intrinsic fees and execution of the transaction, extracted from the `$ggas` register. Converting unspent gas to a fee describes how much "change" is left over from the user's payment; the block producer collects this unspent gas as reward.
+- `gas_to_fee` is a function that converts gas to a concrete fee based on a given gas price
+
+Users wishing to submit transactions can incentivize block producers to include their transactions by providing a higher reward in the form of more unspent gas. This is achieved by specifying a higher gas limit on the transaction.
+
+A naturally occurring result of a variable `gasLimit` is the concept of minimum and maximum fees. The minimum fee is thus the exact fee required to pay the fee balance, while the maximum fee is, then, the minimum fee plus any gas remaining after execution potentially paid to the producer as an incentive:
+
+```py
+min_gas = gas_cost(tx)
+max_gas = min_gas + tx.gasLimit
+min_fee = gas_to_fee(min_gas, tx.gasPrice)
+max_fee = gas_to_fee(max_gas, tx.gasPrice)
+```
+
+The cost of the transaction `cost(tx)` must lie within the range defined by [`min_fee`, `max_fee`]. The definition of `max_gas` illustrates that the delta between minimum fees and maximum fees is the user-defined `gasLimit`. Therefore, `min_fee` is the minimum reward the producer is guaranteed to collect, and `max_fee` is the maximum reward the producer is potentially eligible to collect. In practice, the user is always charged the cost of intrinsic fees. Calculating a conversion from `unspentGas` to an unspent fee describes the reward the producer will collect.
```

Comment:
What do you mean by "intrinsic fees"?

Reply:
I'll add a short section on intrinsic fees. The focus of this document, and the additions this PR makes to it, is outlining validity rules, rather than introductory concepts, so I won't go to deeply into it here.